jmetal-algorithm/src/main/java/org/uma/jmetal/algorithm/multiobjective/nsgaiii/util/ReferencePoint.java

```diff
@@ -101,6 +101,7 @@ public void RemovePotentialMember(S solution) {
     while (it.hasNext()) {
       if (it.next().getLeft() == solution) {
```

Comment:
I notice that it does not remove all the equivalent solutions, but only a single instance (`==` is used, not `.equals(solution)`). Only if the same instance is used several times, then yes it is removed several times. Thus, I would suggest to use `equals` instead of `==`, just in case a different instance happens to be used. It would be an additional fix.

Reply:
While that's true, in `EnvironmentalSelection#execute` we have this block:
```java
{
	this.referencePoints.get(min_rp).AddMember();
	this.referencePoints.get(min_rp).RemovePotentialMember(chosen);
	source.add(chosen);
}
```
So we could potentially be removing a certain instance from ReferencePoint while adding a different one to `source`, so I'm not sure if that's the correct approach here.

Reply:
I don't know, I am not familiar with the algorithm. But situations which require to put several times the same instance in a collection, despite having other instances, is not common. There is, like thread reuse, but because it is rare I think it makes the question relevant. At least, if it has been designed with the idea of having a container with unique instances, then the initial code is correct, although performance could be improved with your fix.

Reply:
I'm a bit concerned changing semantics with this and breaking something unrelated unknowingly; so I'll run a few experiments and report back if it breaks anything.

Reply:
Doesn't seem like it makes a difference, after running a few multi-threaded experiments. I'm switching it to equals

Reply:
Well, it was more a comment for the others, actually. They may know better whether or not such a change makes sense compared to the original algorithm. I did not read the papers, personally. Too lazy for that for now {'^_^}.